internal/teststep/config.go

```diff
@@ -45,6 +45,7 @@ type Config interface {
 	HasProviderBlock(context.Context) (bool, error)
 	HasTerraformBlock(context.Context) (bool, error)
 	Write(context.Context, string) error
+	Append(context.Context, string) Config
```

Comment:
I can't remember, are we guaranteed to always be writing to a temporary test directory / test file? I think we copy them to a temporary directory, but I can't remember the details.

Just want to make sure our eventual changes aren't going to be rewriting/appending to the actual source test configuration files 👍🏻

Reply:
Yes, double-checked now.

File:

https://github.com/hashicorp/terraform-plugin-testing/blob/4a1b9f2cd2c7126226c7d3a54ef30dcde2c53060/internal/teststep/file.go#L87

Directory:

https://github.com/hashicorp/terraform-plugin-testing/blob/4a1b9f2cd2c7126226c7d3a54ef30dcde2c53060/internal/teststep/directory.go#L87